airflow/providers/google/cloud/triggers/dataproc.py

```diff
@@ -43,20 +44,28 @@ def __init__(
         gcp_conn_id: str = "google_cloud_default",
         impersonation_chain: str | Sequence[str] | None = None,
         polling_interval_seconds: int = 30,
+        delete_on_error: bool = True,
     ):
         super().__init__()
         self.region = region
         self.project_id = project_id
         self.gcp_conn_id = gcp_conn_id
         self.impersonation_chain = impersonation_chain
         self.polling_interval_seconds = polling_interval_seconds
+        self.delete_on_error = delete_on_error
 
     def get_async_hook(self):
         return DataprocAsyncHook(
             gcp_conn_id=self.gcp_conn_id,
             impersonation_chain=self.impersonation_chain,
         )
 
+    def get_sync_hook(self):
```

Comment:
We should avoid using sync hook methods in the Triggerer as it would block the triggerer thread.

Reply:
This is to achieve the deletion of the cluster when the CancelledError is raised.